discovery/srv.go

```diff
@@ -68,8 +69,8 @@ func SRVGetCluster(name, dns string, defaultToken string, apurls types.URLs) (st
 				n = fmt.Sprintf("%d", tempName)
 				tempName += 1
 			}
-			stringParts = append(stringParts, fmt.Sprintf("%s=%s%s", n, prefix, tcpAddr.String()))
-			log.Printf("discovery: Got bootstrap from DNS for %s at host %s to %s%s", service, host, prefix, tcpAddr.String())
+			stringParts = append(stringParts, fmt.Sprintf("%s=%s%s", n, prefix, host))
+			log.Printf("discovery: Got bootstrap from DNS for %s at %s%s", service, prefix, host)
```

Comment:
Should drop trailing `.` from `host`

``` go
host = strings.TrimSuffix(host, ".")
```

SRV records may include `.` at the end of its host part, you should drop `.` from end, so it may result a verification error:

```
2015/04/09 10:07:35 etcdserver: could not get cluster response from https://etcd-001.XXX.:2380: Get https://etcd-001.XXX.:2380/members: x509: certificate is valid for ...
```